subprojects/delivery/signer/src/main/kotlin/com/avito/plugin/SignBundleTask.kt

```diff
@@ -1,42 +1,18 @@
 package com.avito.plugin
 
-import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.file.RegularFile
+import org.gradle.api.provider.Property
 import org.gradle.api.tasks.InputFile
-import org.gradle.api.tasks.OutputFile
 import org.gradle.workers.WorkerExecutor
 import java.io.File
 import javax.inject.Inject
 
 public abstract class SignBundleTask @Inject constructor(
     workerExecutor: WorkerExecutor
-) : SignArtifactTask(workerExecutor) {
+) : AbstractSignTask(workerExecutor) {
 
     @get:InputFile
-    public abstract val unsignedFileProperty: RegularFileProperty
+    public abstract val bundleFile: Property<RegularFile>
 
-    @get:OutputFile
-    public abstract val signedFileProperty: RegularFileProperty
-
-    override fun unsignedFile(): File {
-        return unsignedFileProperty.get().asFile
-    }
-
-    override fun signedFile(): File {
-        return signedFileProperty.get().asFile
-    }
-
-    override fun hackForArtifactsApi() {
-        signedFile().copyTo(resultLocation(), overwrite = true)
-    }
-
-    /**
-     * transform API is broken for ArtifactType.Bundle, hacking here to provide expected result
-     * https://issuetracker.google.com/issues/174678813
-     */
-    private fun resultLocation(): File {
-        return signedFile().path
-            .replaceAfterLast("/", unsignedFile().name) // replace bugged `out` file name with original one
-            .replace("/$name", "") // remove `signBundleViaServiceVariant` from path, bugged
-            .let { File(it) }
-    }
+    override fun unsignedFile(): File = bundleFile.get().asFile
```

Comment:
As I understand correctly, this code will inline value when the task is creating
So, I propose to use get() = notation

Reply:
yep, you're right